hashintel/hash
# Use 0.3 staging block hub

## 🌟 What is the purpose of this PR?

HASH is updated to load Block Protocol v0.3 blocks as of #1823, but the production [**Þ _Hub_**](https://blockprotocol.org/hub) will be serving 0.2 blocks until we've got everything ready for 0.3 to release.

In the meantime, to be able to test out 0.3 blocks in HASH, we can use a version of blockprotocol.org built off the `0.3` branch in `blockprotocol`.

This PR does that.

## 🔗 Related links





- [Asana task for restoring the production hub](https://app.asana.com/0/1203358502199087/1203801045329699/f) _(internal)_

## 🔍 What does this change?

- Centralise all the references to the BP URL for loading blocks to a shared `blockProtocolHubOrigin` variable
- Set the value of that variable to `https://blockprotocol-git-03.stage.hash.ai`, which is the Vercel alias for 'deployment built from the tip of `0.3`'
- Use `paragraphBlockComponentId` variable in a couple of places where we were previously repeating the literal string for the paragraph's URL in blockprotocol.org

## 🐾 Next steps

- Revert this when the `0.3` branch of `blockprotocol` is merged into `main`

## ❓ How to test this?
**You may need to clear your `localStorage` value for "user workspace blocks" if you have any issue with the below**

1.  Load a page
1.  Load the code block, which is 0.3-compliant, edit the content, refresh to check that the update call succeeded
1.  To confirm that the BP API works, set `BLOCK_PROTOCOL_API_KEY` to a staging API key (generate one in the linked deployment)
